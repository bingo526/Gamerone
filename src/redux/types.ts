import { RouterState } from 'connected-react-router';
import { AuthState } from './auth/types';
import { ProfileState } from './profile/types';
import { SettingsState } from './settings/types';
import { PostState } from './post/types';
import { PostFormState } from './post-form/types';
import { DialogState } from './dialogs/types';
import { RequestStatusState } from './request-status/types';
import { MiscState } from './misc/types';

export const INIT_STATE = 'root/INIT_STATE';
export const LOAD_STATE = 'root/LOAD_STATE';

export const LOAD_INITIAL_GAMES = 'root/LOAD_INITIAL_GAMES';
export const LOAD_NEXT_GAMES = 'root/LOAD_GAMES';

export const LOAD_EXPERIENCES = 'root/LOAD_EXPERIENCES';
export const LOAD_EXPERIENCE_SUMMARY = 'root/LOAD_EXPERIENCE_SUMMARY';

export interface InitStateAction {
  type: typeof INIT_STATE;
}

export type RootStateActions = InitStateAction;

export interface RootState {
  Auth: AuthState;
  Profile: ProfileState;
  Settings: SettingsState;
  Post: PostState;
  PostForm: PostFormState;
  Dialog: DialogState;
  RequestStatus: RequestStatusState;
  router: RouterState;
  Misc: MiscState;
}
