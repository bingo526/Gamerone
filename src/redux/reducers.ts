import Auth from './auth/reducer';
import Profile from './profile/reducer';
import Settings from './settings/reducer';
import Post from './post/reducer';
import RequestStatus from './request-status/reducer';
import PostForm from './post-form/reducer';
import Dialog from './dialogs/reducer';
import Misc from './misc/reducer';

export default {
  Auth,
  Profile,
  Settings,
  Post,
  PostForm,
  Dialog,
  RequestStatus,
  Misc,
};
