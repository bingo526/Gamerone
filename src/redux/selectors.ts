/**
 * Here comes the combined selectors from different states
 */

import { createSelector } from 'reselect';

import { selectCurrentUserId } from './auth/selectors';
import sort from 'fast-sort';

import {
  selectCurrentProfileUser,
  selectCurrentProfileNetworks,
  selectCurrentProfileSponsors,
  selectCurrentProfileProducts,
  selectCurrentProfileLayout,
  selectProfileGames,
  selectProfileGears,
  selectProfileNowPlaying,
  selectResolvedContent,
  selectResolvedContentId,
  selectProfileLatestAchievements,
  selectProfileGearsResponse,
  selectProfileExperience,
  selectProfileExperienceSummary,
} from './profile/selectors';

import {
  selectSettingsSponsors,
  selectSettingsProducts,
  selectSettingsProfileUser,
  selectProfileSocialNetworks,
  selectLayoutSettings,
  selectSettingsGames,
  selectSettingsGears,
  selectSettingsNowPlaying,
  selectSettingsGearsResponse,
  selectSettingsExperience,
  selectSettingsExperienceSummary,
} from './settings/selectors';

import { selectStatus } from './request-status/selectors';

import { DEFAULT_USER_VISIBILITY } from 'utils/constants';
import { NowPlayingModel } from 'models/NowPlayingModel';
import { selectFeed, selectFeedResponse } from './post/selectors';
import { FEED_DATA, LOAD_FEED_REQUEST } from './post/types';
import { PostModel } from 'models/post';
import { Achievement, Post } from 'interfaces';

/**
 * These selectors below are used to select states effectively based on the currently viewing profile.
 *
 * - If the user is on their OWN profile, they select data from `Settings` state
 * - Otherwise, they select data from `Profile` state
 *
 * Note: We initially started by having separate pages like Settings and Profile (hopefully they make sense what they were for), and now we incorporated Settings and Profile features into one page (although still WIP), but the separated logic behind will remain unchanged (or we will consider to update later)
 */

/**
 * determines whether the currently viewing profile is my profile or not
 */
export const selectIsSelfProfile = createSelector(
  [selectResolvedContentId, selectCurrentUserId],
  (resolvedId, authUserId) => resolvedId === authUserId,
);

export const selectProfileUser = createSelector(
  [selectCurrentProfileUser, selectSettingsProfileUser],
  (profileUser, settingsUser) =>
    settingsUser.id === profileUser?.id ? settingsUser : profileUser,
);

/**
 * resolves social networks to be shown on Profile page
 */
export const selectSocialNetworks = createSelector(
  [
    selectIsSelfProfile,
    selectProfileSocialNetworks,
    selectCurrentProfileNetworks,
  ],
  (isSelfProfile, settingsNetworks, profileNetworks) =>
    isSelfProfile ? settingsNetworks : profileNetworks,
);

/**
 * resolves sponsors to be shown on Profile page
 */
export const selectSponsors = createSelector(
  [selectIsSelfProfile, selectSettingsSponsors, selectCurrentProfileSponsors],
  (isSelfProfile, settingsSponsors, profileSponsors) =>
    isSelfProfile ? settingsSponsors : profileSponsors,
);

/**
 * resolves products to be shown on Profile page
 */
export const selectProducts = createSelector(
  [selectIsSelfProfile, selectSettingsProducts, selectCurrentProfileProducts],
  (isSelfProfile, settingsProducts, profileProducts) =>
    isSelfProfile ? settingsProducts : profileProducts,
);

/**
 * resolves profile layout to be shown on Profile page
 */
export const selectProfileLayout = createSelector(
  [selectIsSelfProfile, selectLayoutSettings, selectCurrentProfileLayout],
  (isSelfProfile, settingsLayout, profileLayout) => {
    const layout = isSelfProfile ? settingsLayout : profileLayout;
    if (layout && !layout.visibility)
      layout.visibility = DEFAULT_USER_VISIBILITY;
    return layout;
  },
);

/**
 * resolves games to be shown on Profile page
 */
export const selectGames = createSelector(
  [selectIsSelfProfile, selectSettingsGames, selectProfileGames],
  (isSelfProfile, settingsGames, profileGames) =>
    isSelfProfile ? settingsGames : profileGames,
);

export const selectGamesStatus = createSelector(
  [selectIsSelfProfile, selectStatus],
  (isSelf, status) =>
    status.get(`${isSelf ? 'settings' : 'profile'}/LOAD_PAGE_REQUEST/games`),
);

export const selectGamesResponse = createSelector(
  [
    selectIsSelfProfile,
    selectSettingsGearsResponse,
    selectProfileGearsResponse,
  ],
  (isOwner, settingsResponse, profileResponse) =>
    isOwner ? settingsResponse : profileResponse,
);

export const selectHasMoreGames = createSelector(
  selectGamesResponse,
  (response) => (response ? response.currentPage >= response.lastPage : false),
);

/**
 * resolves experiences to be shown on Experience page
 */
export const selectExperience = createSelector(
  [selectIsSelfProfile, selectSettingsExperience, selectProfileExperience],
  (isSelfProfile, settingsExperience, profileExperience) =>
    isSelfProfile ? settingsExperience : profileExperience,
);

export const selectExperienceStatus = createSelector(
  [selectIsSelfProfile, selectStatus],
  (isSelf, status) =>
    status.get(`${isSelf ? 'settings' : 'profile'}/GET_EXPERIENCES_REQUEST`),
);

export const selectExperienceSummary = createSelector(
  [
    selectIsSelfProfile,
    selectSettingsExperienceSummary,
    selectProfileExperienceSummary,
  ],
  (isSelfProfile, settingsExperienceSummary, profileExperienceSummary) =>
    isSelfProfile ? settingsExperienceSummary : profileExperienceSummary,
);

export const selectExperienceSummaryStatus = createSelector(
  [selectIsSelfProfile, selectStatus],
  (isSelf, status) =>
    status.get(
      `${
        isSelf
          ? 'settings/GET_USER_ATTRIBUTES_REQUEST'
          : 'profile/GET_EXPERIENCE_SUMMARY_REQUEST'
      }`,
    ),
);

/**
 * resolves gears to be shown on Profile page
 */
export const selectGears = createSelector(
  [selectIsSelfProfile, selectSettingsGears, selectProfileGears],
  (isSelfProfile, settingsGears, profileGears) =>
    isSelfProfile ? settingsGears : profileGears,
);

export const selectGearsStatus = createSelector(
  [selectIsSelfProfile, selectStatus],
  (isSelf, status) =>
    status.get(`${isSelf ? 'settings' : 'profile'}/LOAD_PAGE_REQUEST/gears`),
);

/**
 * resolves now playing data to be shown on Profile page
 */
export const selectNowPlaying = createSelector(
  [selectIsSelfProfile, selectSettingsNowPlaying, selectProfileNowPlaying],
  (isSelfProfile, settingsNowPlaying, profileNowPlaying) =>
    (isSelfProfile ? settingsNowPlaying : profileNowPlaying) ||
    new NowPlayingModel(),
);

/**
 * resolves now playing data to be shown on Profile page
 */
export const selectLatestAchievements = createSelector(
  [selectProfileLatestAchievements],
  (profileLatestAchievements) => {
    if (profileLatestAchievements) {
      return sort(profileLatestAchievements).desc((a: Achievement) => a.date);
    }
    return [];
  },
);

/**
 * Select resolved content type
 *
 * private | user | club | game
 */
export const selectResolvedContentType = createSelector(
  [selectIsSelfProfile, selectResolvedContent],
  (isSelf, content) =>
    (isSelf ? 'private' : content?.contentType || 'private') as FEED_DATA,
);

/**
 * Select currently viewing profile feed whether it's user, game, or club
 */
export const selectProfileFeed = createSelector(
  [selectFeed, selectResolvedContentType, selectResolvedContentId],
  (feedData, contentType, contentId) => {
    if (contentId in feedData[contentType]) {
      return sort([...feedData[contentType][contentId]]).desc(
        (p: Post) => p.createdAt,
      );
    }
    return contentType === 'private'
      ? sort([...feedData.private[0]]).desc((p: Post) => p.createdAt)
      : [];
  },
);

/**
 * Return the rest of the feed excluding the latest post
 */
export const selectProfileFeedRest = createSelector(selectProfileFeed, (feed) =>
  feed.slice(1),
);

export const selectHasMoreProfileFeed = createSelector(
  [selectFeedResponse, selectResolvedContentType, selectResolvedContentId],
  (feedResponse, contentType, contentId) => {
    if (contentId in feedResponse[contentType]) {
      return feedResponse[contentType][contentId]?.offset !== 0;
    }
    return contentType === 'private'
      ? feedResponse.private[0]?.offset !== 0
      : false;
  },
);

export const selectProfileFeedStatus = createSelector(
  [selectStatus, selectResolvedContentType],
  (status, contentType) => status.get(LOAD_FEED_REQUEST + '/' + contentType),
);

export const selectProfileLatestPost = createSelector(
  selectProfileFeed,
  (feed) => (feed.length > 0 ? (feed[0] as PostModel) : new PostModel()),
);
