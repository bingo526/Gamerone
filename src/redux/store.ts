import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import { createBrowserHistory as createHistory } from 'history';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';
import createSagaMiddleware from 'redux-saga';
import reducers from './reducers';
import rootSaga from './sagas';

const history = createHistory();
const sagaMiddleware = createSagaMiddleware();
const routeMiddleware = routerMiddleware(history);
const middlewares = [thunk, sagaMiddleware, routeMiddleware];

const composeEnhancers =
  (window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] as typeof compose) || compose;

const store = createStore(
  combineReducers({
    ...reducers,
    router: connectRouter(history),
  }),
  composeEnhancers(applyMiddleware(...middlewares)),
);
sagaMiddleware.run(rootSaga);
export { store, history };
