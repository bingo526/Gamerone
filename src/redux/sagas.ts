import { selectCurrentUserId } from './auth/selectors';
import {
  all,
  select,
  take,
  race,
  takeLatest,
  put,
  fork,
  call,
} from 'redux-saga/effects';

import authSagas from './auth/saga';
import profileSagas from './profile/saga';
import settingsSagas from './settings/saga';
import postSagas from './post/saga';
import postFormSagas from './post-form/saga';
import miscSagas from './misc/saga';

import {
  LOAD_INITIAL_GAMES,
  LOAD_NEXT_GAMES,
  LOAD_EXPERIENCES,
  LOAD_EXPERIENCE_SUMMARY,
} from './types';
import { selectCurrentProfileId } from './profile/selectors';
import { selectIsSelfProfile } from './selectors';
import {
  CHECK_AUTHORIZATION_SUCCESS,
  CHECK_AUTHORIZATION_ERROR,
} from './auth/types';

import SettingsActions from './settings/actions';
import ProfileActions from './profile/actions';
import { getToken } from 'utils/token';

export function* waitUntilAuthFinishes() {
  const currentUserId = yield select(selectCurrentUserId);
  const token = getToken().get('idToken');
  if (token && currentUserId === 0) {
    yield race({
      success: take(CHECK_AUTHORIZATION_SUCCESS),
      error: take(CHECK_AUTHORIZATION_ERROR),
    });
  }
}

function* loadUserExperienceSummary() {
  yield takeLatest(LOAD_EXPERIENCE_SUMMARY, function* () {
    yield call(waitUntilAuthFinishes);

    const userId = yield select(selectCurrentProfileId);
    const isOwner = yield select(selectIsSelfProfile);

    if (isOwner) {
      yield put(SettingsActions.getUserAttributes());
    } else if (userId) {
      yield put(ProfileActions.getExperienceSummary(userId));
    }
  });
}

function* loadUserExperiences() {
  yield takeLatest(LOAD_EXPERIENCES, function* () {
    yield call(waitUntilAuthFinishes);

    const userId = yield select(selectCurrentProfileId);
    const isOwner = yield select(selectIsSelfProfile);

    if (isOwner) {
      yield put(SettingsActions.getExperiences());
    } else if (userId) {
      yield put(ProfileActions.getExperiences(userId));
    }
  });
}

function* loadInitialGames() {
  yield takeLatest(LOAD_INITIAL_GAMES, function* () {
    yield call(waitUntilAuthFinishes);

    const userId = yield select(selectCurrentProfileId);
    const isOwner = yield select(selectIsSelfProfile);

    if (isOwner) {
      yield put(SettingsActions.loadInitialPage('games'));
    } else if (userId) {
      yield put(ProfileActions.loadInitialPage('games'));
    }
  });
}

function* loadNextGames() {
  yield takeLatest(LOAD_NEXT_GAMES, function* () {
    const userId = yield select(selectCurrentProfileId);
    const isOwner = yield select(selectIsSelfProfile);

    if (isOwner) {
      yield put(SettingsActions.loadNextPage('games'));
    } else if (userId) {
      yield put(ProfileActions.loadNextPage('games'));
    }
  });
}

export default function* rootSaga() {
  yield all([
    fork(loadInitialGames),
    fork(loadNextGames),

    fork(loadUserExperiences),
    fork(loadUserExperienceSummary),

    authSagas(),
    profileSagas(),
    settingsSagas(),
    postSagas(),
    postFormSagas(),
    miscSagas(),
  ]);
}
