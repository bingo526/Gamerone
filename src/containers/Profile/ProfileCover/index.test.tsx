import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createProduct } from 'redux';
import { render, waitForElement } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { act } from 'react-dom/test-utils';
import profileReducer from 'redux/profile/reducer';
import authReducer from 'redux/auth/reducer';
import ProfileHeader from './index';

const testProfile = {
  slug: 'vadim',
  contentType: 'user',
  contentId: 1449,
  content: {
    id: 1449,
    username: 'vadim',
    email: 'vadimas.workteam@gmail.com',
    firstName: 'frontend',
    lastName: 'dev',
    bio: null,
    avatar: 'profile/vadim.svg',
    banner: null,
    playCount: 0,
    gameCount: 0,
    badgeCount: 0,
    followCount: 0,
    followerCount: 0,
    birthDate: null,
  },
};

const renderWithProfileReducer = (
  component = <ProfileHeader />,
  route,
  {
    initialState,
    store = createProduct(profileReducer, { Profile: initialState }),
  } = {},
) => {
  const history = createMemoryHistory();
  if (route) history.push(route);

  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>,
    ),
    history,
  };
};

const renderWithAuthReducer = (
  component = <ProfileHeader />,
  route,
  {
    initialState,
    store = createProduct(authReducer, { Auth: initialState }),
  } = {},
) => {
  const history = createMemoryHistory();
  if (route) history.push(route);

  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>,
    ),
    history,
  };
};

describe('header render on Profile and Home page', () => {
  test('shows username on profile page', async () => {
    const username = 'vadim';

    await act(async () => {
      const { getByLabelText, asFragment } = renderWithProfileReducer(
        <ProfileHeader />,
        '/' + username,
        {
          initialState: { currentProfile: testProfile },
        },
      );

      await waitForElement(() => getByLabelText(/username/));
      expect(asFragment(<ProfileHeader />)).toMatchSnapshot();
      expect(getByLabelText(/username/)).toHaveTextContent(username);
    });
  });

  test('shows username on home page when authenticated', async () => {
    const username = 'vadim';

    await act(async () => {
      const { getByLabelText, asFragment } = renderWithAuthReducer(
        <ProfileHeader />,
        '/',
        {
          initialState: { user: testProfile.content },
        },
      );

      await waitForElement(() => getByLabelText(/username/));
      expect(asFragment(<ProfileHeader />)).toMatchSnapshot();
      expect(getByLabelText(/username/)).toHaveTextContent(username);
    });
  });
});

describe('header elements', () => {
  test('renders username, fullname and avatar', async () => {
    await act(async () => {
      const { firstName, lastName, username } = testProfile.content;
      const {
        getByLabelText,
        asFragment,
        getByAltText,
      } = renderWithProfileReducer(<ProfileHeader />, '/vadim', {
        initialState: { currentProfile: testProfile },
      });

      expect(asFragment(<ProfileHeader />)).toMatchSnapshot();

      // renders full name
      await waitForElement(() => getByLabelText(/fullname/));
      expect(getByLabelText(/fullname/)).toHaveTextContent(
        firstName + ' ' + lastName,
      );

      // renders username
      expect(getByLabelText(/username/)).toHaveTextContent(username);

      // renders avatar
      expect(getByAltText(username)).toBeInTheDocument();
    });
  });

  test('shows club text after username when profile is club', async () => {
    await act(async () => {
      const { getByLabelText, asFragment } = renderWithProfileReducer(
        <ProfileHeader club />,
        '/vadim',
        {
          initialState: {
            currentProfile: { ...testProfile, contentType: 'club' },
          },
        },
      );

      expect(asFragment(<ProfileHeader club />)).toMatchSnapshot();

      // shows club text
      await waitForElement(() => getByLabelText(/username/));
      expect(getByLabelText(/username/)).toHaveTextContent(
        testProfile.content.username + ' (Club)',
      );
    });
  });

  test('should show default placeholder when avatar is empty', async () => {
    await act(async () => {
      const { getByAltText, asFragment } = renderWithProfileReducer(
        <ProfileHeader />,
        '/vadim',
        {
          initialState: {
            currentProfile: {
              ...testProfile,
              content: { ...testProfile.content, avatar: null },
            },
          },
        },
      );

      expect(asFragment(<ProfileHeader />)).toMatchSnapshot();

      // shows club text
      await waitForElement(() => getByAltText('vadim'));

      const avatar = getByAltText('vadim');

      expect(avatar.src).toMatch(/placeholder.svg$/);
    });
  });
});
