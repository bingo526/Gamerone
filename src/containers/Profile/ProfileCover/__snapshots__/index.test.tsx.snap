// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header elements renders username, fullname and avatar 1`] = `
<DocumentFragment>
  <main
    class="container"
    role="main"
  >
    <div
      style="height: 220px; background-image: url(); background-size: cover; background-position: center; background-repeat: no-repeat;"
    >
      <div>
        <div
          class="d-flex"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`header elements should show default placeholder when avatar is empty 1`] = `
<DocumentFragment>
  <main
    class="container"
    role="main"
  >
    <div
      style="height: 220px; background-image: url(); background-size: cover; background-position: center; background-repeat: no-repeat;"
    >
      <div>
        <div
          class="d-flex"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`header elements shows club text after username when profile is club 1`] = `
<DocumentFragment>
  <main
    class="container"
    role="main"
  >
    <div
      style="height: 220px; background-image: url(); background-size: cover; background-position: center; background-repeat: no-repeat;"
    >
      <div>
        <div
          class="d-flex"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`header render on Profile and Home page shows username on home page when authenticated 1`] = `
<DocumentFragment>
  <div
    class="col-md-12"
  >
    <div
      class="card mb-2"
      style="height: 220px; background-image: url(); background-size: cover; background-position: center; background-repeat: no-repeat;"
    >
      <div
        class="card-body"
      >
        <div
          class="d-flex"
        >
          <div
            class="d-flex justify-content-center col-md-3 align-items-center"
            style="margin-top: 120px; padding-left: 0px; margin-left: -15px;"
          >
            <img
              alt="vadim"
              class="avatar xl border border-body border-width-5"
              id="avatar"
              src="https://cdn.g1.gg/128/profile/vadim.svg"
            />
          </div>
          <div
            class="d-flex flex-column flex-fill align-self-center mt-4"
          >
            <h3
              aria-label="username"
            >
              vadim 
            </h3>
            <div
              aria-label="fullname"
            >
              frontend dev
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="d-flex flex-fill justify-content-end"
    style="height: 80px; padding-left: 15px; padding-right: 15px;"
  >
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Monster Energy
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      SAP
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Honda
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Twitch
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Hyper X
    </div>
  </div>
</DocumentFragment>
`;

exports[`header render on Profile and Home page shows username on profile page 1`] = `
<DocumentFragment>
  <div
    class="col-md-12"
  >
    <div
      class="card mb-2"
      style="height: 220px; background-image: url(); background-size: cover; background-position: center; background-repeat: no-repeat;"
    >
      <div
        class="card-body"
      >
        <div
          class="d-flex"
        >
          <div
            class="d-flex justify-content-center col-md-3 align-items-center"
            style="margin-top: 120px; padding-left: 0px; margin-left: -15px;"
          >
            <img
              alt="username"
              class="avatar xl border border-body border-width-5"
              id="avatar"
              src="https://cdn.g1.gg/128/assets/placeholder/avatar.svg"
            />
          </div>
          <div
            class="d-flex flex-column flex-fill align-self-center mt-4"
          >
            <h3
              aria-label="username"
            >
               
            </h3>
            <div
              aria-label="fullname"
            >
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="d-flex flex-fill justify-content-end"
    style="height: 80px; padding-left: 15px; padding-right: 15px;"
  >
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Monster Energy
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      SAP
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Honda
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Twitch
    </div>
    <div
      class="mr-2"
      style="width: 150px;"
    >
      Hyper X
    </div>
  </div>
</DocumentFragment>
`;
